test/integration/builder/deletes.js

```diff
@@ -82,5 +82,86 @@ module.exports = function (knex) {
           );
         });
     });
+    it('#4152 Should allow returns with deletes on tables with triggers', async function () {
+      if (knex.client.driverName !== 'mssql') {
+        return true;
+      }
+
+      return await knex.transaction(async function () {
+        // Reused variables
+        const primaryTable = 'test_return_with_trigger_primary';
+        const secondaryTable = 'test_return_with_trigger_secondary';
+        const primaryLink = 'foreign_id';
+        const secondaryLink = 'looping_id';
+        const triggerName = 'tr_test_return_with_trigger';
+        let insertedId;
+        let deletedId;
+
+        async function setup() {
```

Comment:
it would be more idiomatic for mocha if you extracted setup into `before` block and cleanup into `after` block. Probably will need additional `context` or `describe` block for that.

Reply:
I will not be able to reuse the variables to make maintenance a little bit easier. I would either have to break them into global or just hard code. I am okay doing that just want to make sure first.

Would you like that in a new file or just have two describes in that file?

Reply:
Yeah, I think both new tests could be moved to a new file.

Reply:
Describes can have layers, so you could put them into top-level describe.